TPMainController.m

```diff
@@ -201,8 +204,30 @@ - (void)screenDidLock:(NSNotification*)notification
 	TPClientController * clientController = [TPClientController defaultController];
 	if([clientController isControlling])
 		[clientController stopControl];
+	if ([clientController currentConnection] != nil && [[TPPreferencesManager sharedPreferencesManager] boolForPref:SYNC_LOCK_STATUS]) {
```

Comment:
We would need to iterate over that array and create new connections since `sendMessage` requires a `TPNetworkConnection`. Alternately we could try having TPConnectionsManager hold a list of active connections we could iterate over. `_connections` might be that list, but I haven't tried to unravel how it interacts with `TPUDPSocket`, `TPTCPSocket`, and `TPTCPSecureSocket`.

Reply:
I just took a look. 

https://github.com/johndbritton/teleport/blob/8b2c0ef75cd93391d2ffb8b13a23e6f7a685f351/TPConnectionsManager.m#L259-L270

and

https://github.com/johndbritton/teleport/blob/8b2c0ef75cd93391d2ffb8b13a23e6f7a685f351/TPConnectionsManager.m#L325-L328

Seem to indicate that you're correct, `_connections` is a collection of the connections.